embeddings/pipeline/lightning_pipeline.py

```diff
@@ -13,16 +19,51 @@ class LightningPipeline(
     Pipeline[EvaluationResult],
     Generic[Data, ModelResult, EvaluationResult],
 ):
+    DEFAULT_TASK_TRAIN_KWARGS: Dict[str, Any] = {"devices": "auto", "accelerator": "auto"}
+    DEFAULT_TASK_MODEL_KWARGS: Dict[str, Any] = {"use_scheduler": True}
+    DEFAULT_DATAMODULE_KWARGS: Dict[str, Any] = {"max_seq_length": None}
+    DEFAULT_MODEL_CONFIG_KWARGS: Dict[str, Any] = {"classifier_dropout": None}
+    DEFAULT_LOGGING_KWARGS: Dict[str, Any] = {
+        "use_tensorboard": True,
+        "use_wandb": True,
+        "use_csv": True,
+    }
+    DEFAULT_EARLY_STOPPING_KWARGS: Dict[str, Any] = {
+        "monitor": "val/Loss",
+        "mode": "min",
+        "patience": 3,
+    }
+
     def __init__(
         self,
         datamodule: BaseDataModule[Data],
         model: Model[BaseDataModule[Data], ModelResult],
         evaluator: Evaluator[ModelResult, EvaluationResult],
+        output_path: Path,
+        **kwargs: Any
     ) -> None:
         self.datamodule = datamodule
         self.model = model
         self.evaluator = evaluator
+        self.output_path = output_path
+
+    def run(self, run_name: Optional[str] = None) -> EvaluationResult:
+        self._save_artifacts()
+        model_result = self.model.execute(data=self.datamodule, run_name=run_name)
+        result = self.evaluator.evaluate(model_result)
+        if self.logging_kwargs["use_wandb"]:
+            wandb.log_artifact(
+                str(self.output_path),
+                name="test",
+                type="output",
+            )
+            wandb.finish()
+        return result
```

Comment:
I'd consider wrapping these logging functionalities as separate method to keep `run()` clean, e.g., using pre- and post- hooks as in hps pipelines